src/Apps/Artwork/ArtworkApp.tsx

```diff
@@ -96,12 +97,32 @@ export const ArtworkApp: React.FC<Props> = props => {
     "amber_artwork_visibility_unlisted"
   )
 
+  const partnerOfferVisibilityEnabled = useFeatureFlag(
+    "emerald_partner-offers-to-artwork-page"
+  )
+
   const showUnlistedArtworkBanner =
     artwork?.visibilityLevel == "UNLISTED" && artwork?.partner
 
   const isPrivateArtwork = privateArtworksEnabled && showUnlistedArtworkBanner
 
-  const showExpiredOfferBanner = !!match?.location?.query?.expired_offer
+  const expectedPartnerOfferID = match?.location?.query?.partner_offer_id as
+    | string
+    | undefined
+
+  const partnerOffer = expectedPartnerOfferID
+    ? extractNodes(me?.partnerOffersConnection)[0]
+    : null
+
+  const expectedPartnerOfferNotFound =
+    partnerOfferVisibilityEnabled &&
+    expectedPartnerOfferID &&
+    (!partnerOffer || partnerOffer.internalID !== expectedPartnerOfferID)
+
+  const showExpiredOfferBanner =
+    expectedPartnerOfferNotFound || !!match?.location?.query?.expired_offer
+
+  // show unavailable artwork imperatively if the query param is present
```

Comment:
Just a suggestion: 

We might as well just move all the banner-related code into an `<ArtworkBanners />` component. It's time, I think! 

And then spread in an `ArtworkBanners_viewer` fragment, by updating the existing fragment:

```
fragment ArtworkApp_viewer on Viewer @argumentDefinitions({ artworkID ...}) {
  ...ArtworkBanners_viewer @arguments(artworkID: $artworkID)
  
   artwork(id: $artworkID) {
     // all of the existing fragment code 
   } 
}